hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java

```diff
@@ -151,31 +150,20 @@ public void validateAdminKey(@Nullable Key key) throws PreCheckException {
      */
     public void validateAccountId(@Nullable AccountID accountId) throws PreCheckException {
         validateAccountID(accountId, INVALID_NODE_ACCOUNT_ID);
-        validateFalsePreCheck(!accountId.hasAccountNum() && accountId.hasAlias(), INVALID_NODE_ACCOUNT_ID);
+        validateFalsePreCheck(
+                !requireNonNull(accountId).hasAccountNum() && accountId.hasAlias(), INVALID_NODE_ACCOUNT_ID);
     }
 
     private void validateEndpoint(@NonNull final ServiceEndpoint endpoint, @NonNull final NodesConfig nodesConfig) {
         requireNonNull(endpoint);
         requireNonNull(nodesConfig);
 
         validateFalse(endpoint.port() == 0, INVALID_ENDPOINT);
+        final var addressLen = endpoint.ipAddressV4().length();
+        validateFalse(addressLen == 0 && endpoint.domainName().trim().isEmpty(), INVALID_ENDPOINT);
         validateFalse(
-                endpoint.ipAddressV4().length() == 0
-                        && endpoint.domainName().trim().isEmpty(),
-                INVALID_ENDPOINT);
-        validateFalse(
-                endpoint.ipAddressV4().length() != 0
-                        && !endpoint.domainName().trim().isEmpty(),
-                IP_FQDN_CANNOT_BE_SET_FOR_SAME_ENDPOINT);
+                addressLen != 0 && !endpoint.domainName().trim().isEmpty(), IP_FQDN_CANNOT_BE_SET_FOR_SAME_ENDPOINT);
         validateFalse(endpoint.domainName().trim().length() > nodesConfig.maxFqdnSize(), FQDN_SIZE_TOO_LARGE);
-        validateFalse(endpoint.ipAddressV4().length() != 0 && !isIPv4(endpoint.ipAddressV4()), INVALID_IPV4_ADDRESS);
-    }
-
-    private boolean isIPv4(@NonNull final Bytes ip) {
-        requireNonNull(ip);
-
-        final var part = "(\\d{1,2}|(0|1)\\d{2}|2[0-4]\\d|25[0-5])";
-        final var regex = part + "\\." + part + "\\." + part + "\\." + part;
-        return ip.asUtf8String().matches(regex);
+        validateFalse(addressLen != 0 && addressLen != 4, INVALID_IPV4_ADDRESS);
```

Comment:
This assumes the user will send exactly four bytes for IPV4. I have unit tests for Bytes.wrap("127.0.0.1") as IPV4 and will fail with addressLen !=4. I think we should allow user send any valid IPV4 as this as Bytes.

Reply:
I fixed all these unit tests.

Reply:
Yes, I saw your change on the unit test.  You still assume the user will send exactly four bytes for IPV4.